javascript/javascript1/week3/homework/index.js

```diff
@@ -0,0 +1,199 @@
+  return distance / speed; //for 1km it takes 1/50 hours
+}
+function getHoursAndMinutes(totalTime) {
+  const hours = Math.floor(totalTime);
+  const minutes = Math.ceil((totalTime - hours) * 60); //ceil meth to round up to the larger integer for time it goes to the largest
+  return { hours, minutes };
+}
+function getTravelingTime(travelinfo) {
+  const { speed, destinationDistance, destination } = travelinfo;
+  const totalTime = getTimeInHour(speed, destinationDistance); //function calling another function
+  const { hours, minutes } = getHoursAndMinutes(totalTime); //desctructured
+  return `${hours} hours and  ${minutes} minutes to reach ${destination}`;
+}
+
+const travelTime = getTravelingTime(travelInformation);
+console.log(travelTime);
+
+console.log("**************");
+/* items in array Removal Part 1 */
+const names = [
+  "Peter",
+  "Ahmad",
+  "Yana",
+  "kristina",
+  "Rasmus",
+  "Samuel",
+  "katrine",
+  "Tala",
+];
+
+const nameToRemove = "Ahmad";
+function removeName(name) {
+  return names.filter((item) => item !== name);
+}
+const newNameList = removeName(nameToRemove);
+console.log(newNameList);
+const nameToRemove2 = "Samuel";
+const indexOfNametoBeRemoved = newNameList.indexOf(nameToRemove2);
+console.log(indexOfNametoBeRemoved);
+if (indexOfNametoBeRemoved >= 0) {
+  newNameList.splice(indexOfNametoBeRemoved, 1); // trying also with splice method
+}
+console.log(newNameList);
+
+console.log("**************");
+/* Notes part 4 */
+const notes = [];
+function saveNote(content, id) {
+  notes.push({
+    id: id,
+    content: content,
+  });
+}
+saveNote("Pick up groceries", 1);
+saveNote("Do laundry", 2);
+saveNote("Read a new book", 3);
+saveNote("Study more", 4);
+console.log(notes);
+
+function getNote(id) {
+  if (typeof id !== "number" || typeof id === "undefined") {
+    console.error("Error Id cannot be undefined and id has to be number");
+    return "Error Occured!";
+  } else {
+    return notes.filter((item) => item.id === id); //used filter MEthod
+  }
+}
+const firstNote = getNote(1);
+const secondNote = getNote(2);
+console.log(getNote("2"));
+
+console.log(firstNote);
+console.log(secondNote);
+
+function logOutNotesFormatted() {
+  /* for Each item print id and content */
+  notes.forEach((item) => {
+    const { id, content } = item;
+    console.log(
+      `The note with id:${id}, has the following note Text : ${content}`
+    );
+  });
+}
+logOutNotesFormatted();
+
+console.log("**************");
+/* Activity cheker and show status part 5 */
+
+const activities = [];
+const limit = 100;
+function addActivity(date, activity, duration) {
+  activities.push({
+    date: date,
+    activity: activity,
+    duration: duration,
+  });
+}
+addActivity("23-08-19", "Youtube", 30);
+addActivity("23-08-19", "Tiktok", 40);
+addActivity("23-08-19", "DuoLingo", 50);
+
+console.log(activities);
+
+function showStatus(activities) {
+  if (activities.length == 0 || activities === "undefined") {
```

Comment:
Always use tripple `=`, but in this case you can skip it all together and just use the `!`
```suggestion
  if (!activities.length || activities === "undefined") {
```